subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -64,8 +71,15 @@ private void forwardContent() {
                 if (nread < 0) {
                     break;
                 }
-                outputStream.write(buffer, 0, nread);
-                outputStream.flush();
+                BuildOperationRef startupRef = this.startupRef;
+                if (startupRef != null) {
```

Comment:
Would calling `with` below with a null startupRef mean something else?

Reply:
No. I suppose I could skip the null-check but it seems like unnecessary setting and unsetting of the thread local.